modules/core/CMakeLists.txt

```diff
@@ -168,6 +168,10 @@ if(HAVE_HPX)
   ocv_target_link_libraries(${the_module} LINK_PRIVATE "${HPX_LIBRARIES}")
 endif()
 
+if(WITH_OPENMP AND DEFINED OpenMP_CXX_INCLUDE_DIRS AND OpenMP_CXX_INCLUDE_DIRS)
```

Comment:
Okay, let me bake another commit for this.

Reply:
> Empty OpenMP_CXX_LIBRARIES would trigger CMake error below with missing argument, so check should use OpenMP_CXX_LIBRARIES instead.

I found that `OpenMP_CXX_LIBRARIES` is defined [since CMake>=3.9.0](https://cmake.org/cmake/help/v3.9/module/FindOpenMP.html#backward-compatibility), while `OpenMP_CXX_INCLUDE_DIRS` is defined [since CMake>=3.16.0](https://cmake.org/cmake/help/v3.16/module/FindOpenMP.html#backward-compatibility). So `OpenMP_CXX_LIBRARIES` is never empty when `OpenMP_CXX_INCLUDE_DIRS` has value. It actually behaves exactly like this in my experiences below:

 |              | Linux (Ubuntu 20.04)          |
 | ------------ | ----------------------------- |
 | cmake-3.5.1  | inc: no def, lib: no def      |
 | cmake-3.9.6  | inc: no def, lib: def +value  |
 | cmake-3.15.7 | inc: no def, lib: def + value |
 | cmake-3.16.0 | inc: def, lib: def + value    |